docs/user-guide/reminders-and-external-events.rst

```diff
@@ -0,0 +1,257 @@
+For example, some moisture-sensor attached to a Raspberry Pi should inform your personal assistant that your favourite
+plant needs watering, and your assistant should then relay this message to you.
+
+To do this, your Raspberry Pi needs to send a message to the `trigger_intent endpoint <../../api/http-api/#operation/triggerConversationIntent>`_ of your conversation.
+As the name says, this injects a user intent (possibly with entities) into your conversation.
+So for Rasa it is almost as if you had entered a message that got classified with this intent and these entities.
+Rasa then needs to respond to this input with an action such as ``action_warn_dry``.
+The easiest and most reliable way to connect this action with the intent is via the :ref:`mapping-policy`.
+
+
+.. _getting-conversation-id:
+
+Getting the conversation ID
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The first thing we need is the Session ID of the conversation that your sensor should send a notification to.
+An easy way to get this is to define a custom action (see :ref:`custom-actions`) that displays the ID in the conversation.
+For example:
+
+.. code-block:: python
+
+  class ActionTellID(Action):
+    """Informs the user about the conversation ID."""
+
+    def name(self) -> Text:
+        return "action_tell_id"
+
+    def run(
+        self, dispatcher, tracker: Tracker, domain: Dict[Text, Any]
+    ) -> List[Dict[Text, Any]]:
+
+        conversation_id = tracker.sender_id
+
+        dispatcher.utter_message(
+            f"The ID of this conversation is: '{conversation_id}'."
+        )
+
+        return []
+
+
+In addition, we also declare an intent ``ask_id``, define some NLU data for it, and add both ``action_tell_id`` and ``ask_id`` to the domain file, where we specify that one should trigger the other:
+
+.. code-block:: md
+
+  intents:
+    - ask_id:
+      triggers: action_tell_id
+
+
+Now, when you ask "What is the ID of this conversation?", the assistant replies with something like "The ID of this conversation is: 38cc25d7e23e4dde800353751b7c2d3e".
+See the ``reminderbot`` example project under ``rasa/examples/reminderbot`` for details.
+
+
+.. _responding_to_external_events:
+
+Responding to external events
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Now that we have our Session ID, we need to prepare the assistant so it responds to messages from the sensor.
+To this end, we define a new intent ``EXTERNAL_dry_plant`` without any NLU data.
+This intent will later be triggered by the external sensor.
+Here, we start the intent name with ``EXTERNAL_`` to indicate that this is not something the user would say, but you can name the intent however you like.
+
+In the domain file, we now connect the intent ``EXTERNAL_dry_plant`` with another custom action ``action_warn_dry``, e.g.
+
+.. code-block:: python
+
+  class ActionWarnDry(Action):
+    """Informs the user that a plant needs water."""
+    def name(self) -> Text:
+        return "action_warn_dry"
+
+    def run(
```

Comment:
```suggestion
    async def run(
```

Reply:
This causes `No registered action found for name 'action_set_reminder'.`, and I just pulled the latest `rasa_sdk`. We should do this for all example bots in a separate PR.